src/exchanges/coinex-client.spec.js

```diff
@@ -0,0 +1,181 @@
+const Coinex = require("./coinex-client");
+jest.mock("winston", () => ({ info: jest.fn(), warn: jest.fn(), error: jest.fn() }));
+
+let client;
+let market1 = {
+  id: "BTCBCH",
+  base: "BCH",
+  quote: "BTC",
+};
+
+let market2 = {
+  id: "LTCBTC",
+  base: "BTC",
+  quote: "LTC",
+};
+
+beforeAll(() => {
+  client = new Coinex();
+});
+
+test("it should support tickers", () => {
+  expect(client.hasTickers).toBeTruthy();
+});
+
+test("it should support trades", () => {
+  expect(client.hasTrades).toBeTruthy();
+});
+
+test("it should not support level2 snapshots", () => {
+  expect(client.hasLevel2Snapshots).toBeFalsy();
+});
+
+test("it should support level2 updates", () => {
+  expect(client.hasLevel2Updates).toBeTruthy();
+});
+
+test("it should not support level3 snapshots", () => {
+  expect(client.hasLevel3Snapshots).toBeFalsy();
+});
+
+test("it should not support level3 updates", () => {
+  expect(client.hasLevel3Updates).toBeFalsy();
+});
+
+test("should subscribe and emit ticker events", done => {
+  client.subscribeTicker(market1);
+  client.on("ticker", function tickerHandler(ticker) {
+    expect(ticker.fullId).toMatch("Coinex:BCH/BTC");
+    expect(ticker.timestamp).toBeGreaterThan(1531677480465);
+    expect(typeof ticker.last).toBe("string");
+    expect(typeof ticker.open).toBe("string");
+    expect(typeof ticker.high).toBe("string");
+    expect(typeof ticker.low).toBe("string");
+    expect(typeof ticker.volume).toBe("string");
+    expect(typeof ticker.quoteVolume).toBe("string");
+    expect(typeof ticker.change).toBe("string");
+    expect(typeof ticker.changePercent).toBe("string");
+    expect(typeof ticker.bid).toBe("undefined");
+    expect(typeof ticker.bidVolume).toBe("undefined");
+    expect(typeof ticker.ask).toBe("undefined");
+    expect(typeof ticker.askVolume).toBe("undefined");
+    expect(parseFloat(ticker.last)).toBeGreaterThan(0);
+    expect(parseFloat(ticker.open)).toBeGreaterThan(0);
+    expect(parseFloat(ticker.high)).toBeGreaterThan(0);
+    expect(parseFloat(ticker.low)).toBeGreaterThan(0);
+    expect(parseFloat(ticker.volume)).toBeGreaterThan(0);
+    expect(parseFloat(ticker.quoteVolume)).toBeGreaterThan(0);
+    expect(Math.abs(parseFloat(ticker.change))).toBeGreaterThan(0);
+    expect(Math.abs(parseFloat(ticker.changePercent))).toBeGreaterThan(0);
```

Comment:
Minor change:  This is actually a bug in other tests and a cause of a lot of failed tests. change and  changePercent should do a check that it is not NaN.  I think this should be .not.isFalsey().  I'm adding an issue to fix this in other files as well.

Reply:
BTW add issue #23 for fixing this in other tests.